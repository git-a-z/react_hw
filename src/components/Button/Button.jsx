import React from 'react'
import Button from '@mui/material/Button';
import ListItemIcon from '@mui/material/ListItemIcon';
import SendIcon from '@mui/icons-material/Send';

export function MyButton(props) {
    return (
        <Button variant="contained"
            {...props}>
            {props.children}
        </Button>
    )
}

export function MySendButton(props) {
    return (
        <Button variant="contained"
            {...props}>
            <ListItemIcon>
                <SendIcon className="Text" />
            </ListItemIcon>
            {props.children}
        </Button>
    )
}
