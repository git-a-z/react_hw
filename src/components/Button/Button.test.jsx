import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import { MyButton, MySendButton } from './Button'

describe('MyButton', () => {
    it('render component Button', () => {
        render(<MyButton>Change Name</MyButton>)
    })

    it('render with snapshot', () => {
        const { asFragment } = render(<MyButton>Change name</MyButton>)
        expect(asFragment()).toMatchSnapshot()
    })

    it('render component with text name button', () => {
        render(<MyButton>Change name</MyButton>)
        expect(screen.getByText(/Change name/)).toBeInTheDocument()
    })

    it('render multiple components', () => {
        render(
            <>
                <MyButton>Edit</MyButton>
                <MyButton>Delete</MyButton>
            </>
        )
        expect(screen.queryAllByRole('button').length).toBe(2)
    })

    it('button disabled', () => {
        render(<MyButton disabled>Delete</MyButton>)
        expect(screen.getByText(/Delete/)).toBeDisabled()
    })

    it('test example', async () => {
        const onChange = jest.fn()
        render(<input type="checkbox" onChange={onChange} />)

        const checkbox = screen.getByRole('checkbox')
        await userEvent.dblClick(checkbox)
        expect(onChange).toHaveBeenCalledTimes(2)
        expect(checkbox).not.toBeChecked()
    })
})

describe('MySendButton', () => {
    it('render component Button', () => {
        render(<MySendButton>Send</MySendButton>)
    })

    it('render with snapshot', () => {
        const { asFragment } = render(<MySendButton>Send</MySendButton>)
        expect(asFragment()).toMatchSnapshot()
    })

    it('render component with text name button', () => {
        render(<MySendButton>Send</MySendButton>)
        expect(screen.getByText(/Send/)).toBeInTheDocument()
    })

    it('render multiple components', () => {
        render(
            <>
                <MySendButton>Edit</MySendButton>
                <MySendButton>Delete</MySendButton>
            </>
        )
        expect(screen.queryAllByRole('button').length).toBe(2)
    })

    it('button disabled', () => {
        render(<MySendButton disabled>Delete</MySendButton>)
        expect(screen.getByText(/Delete/)).toBeDisabled()
    })
})
