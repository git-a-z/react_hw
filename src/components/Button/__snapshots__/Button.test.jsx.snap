// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Change name
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`MySendButton render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Text css-i4bv87-MuiSvgIcon-root"
        data-testid="SendIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
        />
      </svg>
    </div>
    Send
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
